Plutarch/Builtin.hs

```diff
@@ -28,8 +32,24 @@ import PlutusTx (Data)
 
 data PBuiltinPair (a :: k -> Type) (b :: k -> Type) (s :: k)
 
+pfstBuiltin :: Term s (PBuiltinPair a b :--> a)
+pfstBuiltin = phoistAcyclic $ pforce . pforce . punsafeBuiltin $ PLC.FstPair
+
+psndBuiltin :: Term s (PBuiltinPair a b :--> b)
+psndBuiltin = phoistAcyclic $ pforce . pforce . punsafeBuiltin $ PLC.SndPair
+
+{- | Construct a builtin pair of 'PData' elements.
+
+Uses 'PAsData' to preserve more information about the underlying 'PData'.
+-}
+ppairDataBuiltin :: Term s (PAsData a :--> PAsData b :--> PBuiltinPair (PAsData a) (PAsData b))
```

Comment:
We probably want to fix the `IsData` instance for pairs then to make it use `PAsData`.

Reply:
Not sure what you meant here. The `IsData` instance is just for `(Integer, [Data])` right? The 2 members aren't `Data` values so we can't use `PAsData` there.

Reply:
You are right.

Reply:
However, why do you need this then?